### ISSUE TITLE
How to run example_PD.pl code parallelly?

### ISSUE THREAD

**Issue by Otoliths** [shawnlaffan/biodiverse#824]
Hi @shawnlaffan,

The calculating speed of this example [example_PD.pl](https://github.com/shawnlaffan/biodiverse/blob/master/example_scripts/example_PD.pl) is time-consuming when loading big data, especially SPATIAL ANALYSIS and MATRIX AND CLUSTER ANALYSES. How to run example_PD.pl code parallelly via Parallel::ForkManager? Whether the following code modification is correct？

- Line133-173
```
###### SPATIAL ANALYSIS
use Parallel::ForkManager;
$pm = Parallel::ForkManager->new($MAX_processes)
if ($do_spatial) {
    foreach my $tree_ref ($tree) {  #  dirty hack - should just use directly since we have only one tree
        my $pid = $pm->start and next;
        #my $name = $spatial_output_prefix . $tree_ref->get_param ('NAME');
		my $name = $input_bds_file;
        my $output = $bd->add_spatial_output (name => $name);
        my $success = eval {
            $output->run_analysis (
                spatial_conditions => $spatial_conditions,
                calculations       => $analyses_to_run,
                tree_ref           => $tree_ref,
            );
       $pm->finish;
        };
        croak $EVAL_ERROR if $EVAL_ERROR;

        if ($success) {  #  export to CSV using defaults
            $output->export (
                file   => $name . '.csv',
                format => 'Delimited text',
                list   => 'SPATIAL_RESULTS',
            );
			#$output->export ( # export asci grids
			#   file   => $name . '.asc',
			#   format => 'ArcInfo asciigrid files',
			#	list   => 'SPATIAL_RESULTS',
            #);
        }

        if (not $retain_outputs) {
            $bd->delete_output (output => $output);
        }
    }
};

```

- Line174-212

```
###### MATRIX AND CLUSTER ANALYSES
###  add a cluster analysis for each tree, using the same spatial params for each
use Parallel::ForkManager;
$pm = Parallel::ForkManager->new($MAX_processes)
if ($do_matrix_cluster) {

    foreach my $tree_ref ($tree) {
       my $pid = $pm->start and next;
        my $name_clust = $matrix_cluster_output_prexif . $tree_ref->get_param ('NAME');
        my $output = $bd->add_cluster_output (name => $name_clust);
        my $success = eval {
            $output->run_analysis (
                spatial_conditions   => $spatial_conditions_clust,
                #spatial_calculations => $analyses_to_run,  #  run these analyses for each node on the tree - comment out if not needed
                tree_ref             => $tree_ref,
                index                => 'S2',  #  this isn't publicly available yet (2011-01-30)
                linkage_function     => 'link_average',
            );
        $pm->finish;
        };

        if ($success) {
             to export the cluster tree
            $output->export (file => $output->get_param ('NAME') . '.nex',
                              format => 'Nexus');  #  export cluster tree to Nexus using defaults

            #  export sparse matrix format to CSV
            $output->export (
                file   => $output->get_param ('NAME') . '.csv',
                format => 'Matrices', type => 'sparse',
            );

        }
        if (not $retain_outputs) {
            $bd->delete_output (output => $output);
        }
    }
};

if ($save_basedata) {
    $bd->save (filename => $output_bds_file);
};

```

**Comment by shawnlaffan** [comment]
Hello @Otoliths 

Parallelisation is certainly of interest and is on my list of things to do.  The level you are proposing is about right, i.e. at the top of the process rather than deep in the internals.  

Are you able to work up a pull request?  

It's also worth checking out the [MCE](https://metacpan.org/dist/MCE/view/lib/MCE.pod) module for another approach, but I have no particular preference so long as code works across platforms.  


Shawn.

**Comment by shawnlaffan** [comment]
This was fixed in #826
